File: day01/day01.py
left_row = []
right_row = []

with open("input.txt", "r") as file:
    for line in file:
        row = line.replace("\n", "").split("   ")
        left_row.append(int(row[0]))
        right_row.append(int(row[1]))

left_row.sort()
right_row.sort()

difference = 0
for entry in range(len(left_row)):
    difference += abs(left_row[entry] - right_row[entry])

print("Part 1:", difference)

total_similarity = 0
for left_side in left_row:
    for right_side in right_row:
        if left_side == right_side:
            total_similarity += left_side

print("Part 2:", total_similarity)

File: day01/day01.js
const fs = require('fs');
const input = String(fs.readFileSync('input.txt'));

const leftAndRight = input.split('\n');

let array1 = [];
let array2 = [];

for (line of leftAndRight) {
  array1.push(Number(line.split('   ')[0]));
  array2.push(Number(line.split('   ')[1]));
}

array1.sort(function (a, b) {
  return a - b;
});

array2.sort(function (a, b) {
  return a - b;
});

let difference = 0;
for (let index = 0; index < array1.length; index++) {
  difference += Math.abs(array1[index] - array2[index]);
}

console.log('Part 1:', difference);

let totalSimilary = 0;
for (leftSide of array1) {
  for (rightSide of array2) {
    if (leftSide === rightSide) {
      totalSimilary += leftSide;
    }
  }
}

console.log('Part 2:', totalSimilary);
